src/services/completions.ts

```diff
@@ -1555,15 +1566,15 @@ function getEntryForMemberCompletion(
     position: number,
     contextToken: Node | undefined,
     formatContext: formatting.FormatContext | undefined,
-): { insertText: string, isSnippet?: true, importAdder?: codefix.ImportAdder, replacementSpan?: TextSpan } {
+): { insertText: string, filterText?: string, isSnippet?: true, importAdder?: codefix.ImportAdder, eraseRange?: TextRange } | undefined {
     const classLikeDeclaration = findAncestor(location, isClassLike);
     if (!classLikeDeclaration) {
-        return { insertText: name };
+        return undefined; // This should never happen.
     }
 
     let isSnippet: true | undefined;
-    let replacementSpan: TextSpan | undefined;
     let insertText: string = name;
+    const filterText: string = name;

```

Comment:
If filter text is `name` why cant editors use `name` itself to sort.. may need to consider`kind` along with that but it feels like we are returning same data.

Reply:
The editor doesn't know that it should use `name` as the `filterText`, because in theory `filterText` could be anything. We could change the TS vscode extension to use the `name` as the `filterText` for class member snippet completions, but then that change would be restricted to vscode, and other editors would potentially run into the problem this PR is trying to fix.